brownie/network/transaction.py

```diff
@@ -278,7 +278,18 @@ def wait(self, required_confs: int) -> None:
                     time.sleep(0.5)
             self._await_confirmation(tx, required_confs)
 
+    def _cover_segment(self, segment):
+        """We run a function for every single trace so coverage could count it."""
+
     def _raise_if_reverted(self, exc: Any) -> None:
+
+        # Should check if need coverage before doing this. Also this
+        # might not be reliable indication of the end of a transaction
+        # lifecycle
+        self._expand_trace()
+        for segment in self.trace:
+            self._cover_segment(segment)
```

Comment:
The actual coverage eval happens in `TransactionReceipt._expand_trace`, and the logic already involves iterating `self.trace`.  I think you could probably hook in there instead, and make it more efficient by only calling `_cover_segment` on trace steps that touch a new statement / branch.